lib/pbench/server/api/resources/datasets_compare.py

```diff
@@ -128,21 +128,27 @@ def _get(self, params: ApiParams, req: Request, context: ApiContext) -> Response
         stream_file = {}
         for dataset in datasets:
             try:
-                info = cache_m.get_inventory(dataset.resource_id, "result.csv")
-                file = info["stream"].read().decode("utf-8")
-                info["stream"].close()
+                file = cache_m.get_inventory_bytes(dataset.resource_id, "result.csv")
+            except CacheManagerError as e:
+                raise APIAbort(
+                    HTTPStatus.BAD_REQUEST,
+                    "unable to extract postprocessed data from {dataset.name}",
+                ) from e
             except Exception as e:
                 raise APIInternalError(
-                    f"{dataset.name} is missing 'result.csv' file"
+                    f"Unexpected error extracting postprocessed data from {dataset.name}"
                 ) from e
             stream_file[dataset.name] = file
 
-        quisby_response = QuisbyProcessing().compare_csv_to_json(
-            benchmark_type, InputType.STREAM, stream_file
-        )
-        if quisby_response["status"] != "success":
-            raise APIInternalError(
-                f"Quisby processing failure. Exception: {quisby_response['exception']}"
+        try:
+            quisby_response = QuisbyProcessing().compare_csv_to_json(
+                benchmark_type, InputType.STREAM, stream_file
             )
-        quisby_response["benchmark"] = benchmark
-        return jsonify(quisby_response)
+            if quisby_response["status"] != "success":
+                raise APIInternalError(
+                    f"Comparison processing failure. Exception: {quisby_response['exception']}"
+                )
+            quisby_response["benchmark"] = benchmark
+            return jsonify(quisby_response)
+        except Exception as e:
+            raise APIInternalError(f"Comparison failed with {str(e)!r}")
```

Comment:
If we raise the `APIInternalError` at line 148, then we're going to catch it and wrap it in another `APIInternalError` exception at line 153-154...is that really what we want?

I have no problem with raising an exception at line 148, but it seems like we should raise something other than `APIInternalError`.  (Just raise an `Exception` if nothing else (like `RuntimeError`) seems better.)

Ditto line 91 in `lib/pbench/server/api/resources/datasets_visualize.py`.

Reply:
Ugh. I really wanted to get this out before I quit yesterday, and I clearly rushed it a bit too much. 😦